From: Yijing Wang <wangyijing@huawei.com>
Date: Tue, 18 Jun 2013 16:12:37 +0800
Subject: [PATCH v2 06/31] bnx2: use pdev->pm_cap instead of
 pci_find_capability(.., PCI_CAP_ID_PM)
Patch-mainline: v3.11-rc1

Git-commit: 85768271315bd368c86a0783bf0c85019afbabec
References: bsc#908686 FATE#317538


Pci core has been saved pm cap register offset by pdev->pm_cap in pci_pm_init()
in init path. So we can use pdev->pm_cap instead of using
pci_find_capability(pdev, PCI_CAP_ID_PM) for better performance and simplified code.

Signed-off-by: Yijing Wang <wangyijing@huawei.com>
Cc: Michael Chan <mchan@broadcom.com>
Cc: netdev@vger.kernel.org
Cc: linux-kernel@vger.kernel.org
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Cho, Yu-Chen <acho@suse.com>
---
 drivers/net/ethernet/broadcom/bnx2.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/broadcom/bnx2.c
+++ b/drivers/net/ethernet/broadcom/bnx2.c
@@ -8101,7 +8101,7 @@ bnx2_init_board(struct pci_dev *pdev, st
 
 	pci_set_master(pdev);
 
-	bp->pm_cap = pci_find_capability(pdev, PCI_CAP_ID_PM);
+	bp->pm_cap = pdev->pm_cap;
 	if (bp->pm_cap == 0) {
 		dev_err(&pdev->dev,
 			"Cannot find power management capability, aborting\n");
